lib/matplotlib/ticker.py

```diff
@@ -973,6 +1078,87 @@ def format_eng(self, num):
+    Percentages are computed as ``x / mx * 100``. So if the data is
+    already scaled to be percentages, `mx` will be 100. Another common
+    situation is where `max` is 1.0.
+    """
+    def __init__(self, mx=100, decimals=None, symbol='%'):
+        """
+        Initializes the formatter.
+
+        `max` is the data value that corresponds to 100%. `symbol` is
+        a string which will be appended to the label. It may be `None`
+        or empty to indicate that no symbol should be used. `decimals`
+        is the number of decimal places to place after the 
+        """
+        self.max = mx + 0.0
+        self.decimals = decimals
+        self.symbol = symbol
+
+    def __call__(self, x, pos=None):
+        """
+        Formats the tick as a percentage with the appropriate scaling.
+        """
+        xmin, xmax = self.axis.get_view_interval()
+        d = abs(xmax - xmin)
+
+        return self.fix_minus(format_pct(x, d))
+
+    def format_pct(self, x, d):
+        """
+        Formats the number as a percentage number with the correct
+        number of decimals and adds the percent symbol, if any.
+
+        If `self.decimals` is `None`, the number of digits after the
+        decimal point is set based on the width of the domain `d` as
+        follows:
+
+        +-------+----------+------------------------+
+        |   d   | decimals |          sample        |
+        +-------+----------+------------------------+
+        + >50   |   0      | ``x = 34.5`` => 34%    |
+        +-------+----------+------------------------+
+        | >5    |   1      | ``x = 34.5`` => 34.5%  |
+        +-------+----------+------------------------+
+        | >0.5  |   2      | ``x = 34.5`` => 34.50% |
+        +-------+----------+------------------------+
+        |  ...  |   ...    |          ...           |
+        +-------+----------+------------------------+
+
+        This method will not be very good for tiny ranges or extremely
+        large ranges. It assumes that the values on the chart are
+        percentages displayed on a reasonable scale.
+        """
+        x = self.convert_to_pct(x)
+        if self.decimals is None:
+            # Luckily Python's built-in `ceil` rounds to +inf, not away
+            # from zero. This is very important since the equation for
+            # `decimals` starts out as `d > 0.5 * 10**(2 - decimals)`
+            # and ends up with `decimals > 2 - log10(2 * d)`.
+            d = self.convert_to_pct(d) # d is a difference, so this works fine
+            decimals = math.ceil(2.0 - math.log10(2.0 * d))
+            if decimals > 5:
+                decimals = 5
+            elif decimals < 0:
+                decimals = 0
+        else:
+            decimals = self.decimals
+        s = '{x:0.{decimals}f}'.format(x=x, decimals=decimals)
+
+        if self.symbol:
+            return s + self.symbol
+        return s
+
+    def convert_to_pct(self, x):
+        return x / self.max * 100.0
```

Comment:
I like to use parentheses to make expressions like this clearer; or multiply by 100 first.